bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/GatewayWebTargets.java

```diff
@@ -0,0 +1,523 @@
+        if (!isRegistered) {
+            invoke(HttpMethod.PUT, register, null, null);
+            isRegistered = true;
+        }
+        return isRegistered;
+    }
+
+    /**
+     * Get hub properties.
+     *
+     * @return a map containing the hub properties.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public Map<String, String> getInformation() throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, info, null, null);
+        try {
+            Info result = jsonParser.fromJson(json, Info.class);
+            if (result == null) {
+                throw new HubProcessingException("getInformation(): missing response");
+            }
+            return Map.of( //
+                    Thing.PROPERTY_FIRMWARE_VERSION, result.getFwVersion(), //
+                    Thing.PROPERTY_SERIAL_NUMBER, result.getSerialNumber());
+        } catch (JsonParseException e) {
+            throw new HubProcessingException("getFirmwareVersions(): JsonParseException");
+        }
+    }
+
+    /**
+     * Get the list of scenes.
+     *
+     * @return the list of scenes.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public List<Scene> getScenes() throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, scenes, null, null);
+        try {
+            return List.of(jsonParser.fromJson(json, Scene[].class));
+        } catch (NullPointerException e) {
+            throw new HubProcessingException("getScenes() missing response");
+        } catch (JsonParseException e) {
+            throw new HubProcessingException("getScenes() JsonParseException");
+        }
+    }
+
+    /**
+     * Get the data for a single shade.
+     *
+     * @param shadeId the id of the shade to get.
+     * @return the shade.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public Shade getShade(int shadeId) throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, String.format(shadeSingle, shadeId), null, null);
+        try {
+            Shade result = jsonParser.fromJson(json, Shade.class);
+            if (result == null) {
+                throw new HubProcessingException("getShade() missing response");
+            }
+            return result;
+        } catch (JsonParseException e) {
+            throw new HubProcessingException("getShade() JsonParseException");
+        }
+    }
+
+    /**
+     * Get the list of shades.
+     *
+     * @return the list of shades.
+     * @throws HubProcessingException if any error occurs.
+     */
+    public List<Shade> getShades() throws HubProcessingException {
+        String json = invoke(HttpMethod.GET, shades, null, null);
+        try {
+            return List.of(jsonParser.fromJson(json, Shade[].class));
+        } catch (NullPointerException e) {
+            throw new HubProcessingException("getScenes() missing response");
```

Comment:
You should not catch `NullPointerException`, but instead avoid it.

Reply:
Ditto.